docs/src/nodes/detailed-hydropower.md

```diff
@@ -0,0 +1,780 @@
+These variabels are hence created for [`HydroGenerator`](@ref) nodes.
+
+- ``\texttt{discharge\_segments}[n, t, q]``: One discharge variable is defined for each segment `q` of the PQ-curve defined by the field `pq_curve` of node ``n`` in operational period ``t`` with unit volume per time unit.\
+  If [`PqPoints`](@ref) are provided, the number of discharge segments will be ``Q``, where ``Q+1`` is the length of the vectors in the fields of [`PqPoints`](@ref).
+  There is only one discharge segment if an energy equivalent is used.
+  The variables ``\texttt{discharge\_segments}`` define the utilisation of each discharge segment and sum up to the total discharge.
+
+  !!! warning "discharge_segments"
+      Sequential allocation is not enforced by binary variables, but allocation will occure sequentially if the problem if set up correctly.
+      Penalties for spilling water, a non-concave PQ-curve or an otherwise non-convex problem are examples thay may result in a non-sequential allocation.
+
+The following variables are created if required by the *[additional constraints](@ref nodes-det_hydro_power-phil-con)*:
+
+- ``\texttt{gen\_penalty\_up}[n, t, p]``: Variable for penalizing violation of the maximum constraint of the resource `p` in direction *up* in `HydroGenerator` node ``n`` in operational period ``t`` with unit volume per time unit.\
+  *Up* implies in this case that the electricity generation is larger than planned.
+- ``\texttt{gen\_penalty\_down}[n, t, p]``: Variable for penalizing violation of the maximum constraint of the resource `p` in direction *down* in `HydroGenerator` node ``n`` in operational period ``t`` with unit volume per time unit.\
+  *Down* implies in this case that the electricity generation is smaller than planned.
+
+#### [Constraints](@id nodes-hydro_generator-math-con)
+
+In the following sections the vector of [`HydroGenerator`](@ref) nodes are omitted from the descriptions.
+Instead, it is implicitly assumed that the constraints are valid ``\forall n ∈ N`` for all [`HydroGenerator`](@ref) types if not stated differently.
+In addition, all constraints are valid ``\forall t \in T`` (that is in all operational periods) or ``\forall t_{inv} \in T^{Inv}`` (that is in all strategic periods).
+
+##### [Standard constraints](@id nodes-hydro_generator-math-con-stand)
+
+[`HydroGenerator`](@ref) nodes utilize in general the standard constraints described in *[Constraint functions for `NetworkNode`](@extref EnergyModelsBase nodes-network_node-math-con)*.
+The majority of these constraints are hence ommitted in the following description.
+
+The function `constraints_capacity` rquires a new method to account for the included [PQ-curve](@ref nodes-det_hydro_power-phil-pq):
+
+```math
+\begin{aligned}
+    \texttt{cap\_use}&[n, t] \leq \texttt{cap\_inst}[n, t] \times \frac{P^{max}}{capacity(n, t)}\\
```

Comment:
Renamed `max_power` function to `max_normalized_power` since this was a more precise name. Documentation has been updated such that it aligns with code. The possibility to use capacity either for discharge or power has also been elaborated: 6fadd616cdb313f80b83b715b74d2365908bb0db